Sources/SpeziLocalLLM/LLMLlama+Generation.swift

```diff
@@ -0,0 +1,167 @@
+//
+// This source file is part of the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import llama
+
+
+/// Extension of ``LLMLlama`` handling the text generation.
+extension LLMLlama {
+    /// Typealias for the llama.cpp `llama_token`.
+    typealias LLMLlamaToken = llama_token
+    
+    
+    /// Based on the input prompt, generate the output with llama.cpp
+    ///
+    /// - Parameters:
+    ///   - prompt: The input `String` prompt.
+    ///   - continuation: A Swift `AsyncThrowingStream` that streams the generated output.
+    func _generate( // swiftlint:disable:this identifier_name function_body_length
+        prompt: String,
+        continuation: AsyncThrowingStream<String, Error>.Continuation
+    ) {
+        self.state = .generating
+        
+        let tokens = tokenize(text: prompt)
+        let nKVReq = UInt32(tokens.count) + UInt32((self.parameters.nLength - Int(tokens.count)))
+        
+        let context = llama_new_context_with_model(model, self.contextParameters.wrapped)
+        guard context != nil else {
+            print("Failed to initialize context")
```

Comment:
I think it might make sense to use Swift Logger here instead of print statement, especially when it comes down to use it in a production system: https://developer.apple.com/documentation/os/logger